packages/web3-utils/src/socket_provider.ts

```diff
@@ -199,7 +200,10 @@ export abstract class SocketProvider<
 	 * @param type - The event type to listen for
 	 * @param callback - The callback to be invoked when the event is emitted
 	 */
-	public on<T = JsonRpcResult>(type: EventType, callback: Web3ProviderEventCallback<T>): void {
+	public on<T = JsonRpcResult>(
+		type: EventType,
+		callback: Web3Eip1193ProviderEventCallback<T> | Web3ProviderEventCallback<T>,
+	): void {
 		this._eventEmitter.on(type, callback);
 	}
```

Comment:
I am thinking that having only one implementation for `on` will not force the eact type restrictions we have at `Ẁeb3BaseProvider`. What do you think about having here something like this?:

```suggestion
	public abstract on(
		type: 'disconnect',
		listener: Web3Eip1193ProviderEventCallback<ProviderRpcError>,
	): void {
		this._eventEmitter.on(type, listener);
	}
	public abstract on<T = JsonRpcResult>(
		type: 'message' | string,
		listener: Web3Eip1193ProviderEventCallback<ProviderMessage> | Web3ProviderEventCallback<T>,
	): void {
		this._eventEmitter.on(type, listener);
	}
	public abstract on(
		type: 'connect',
		listener: Web3Eip1193ProviderEventCallback<ProviderConnectInfo>,
	): void {
		this._eventEmitter.on(type, listener);
	}
	public abstract on(
		type: 'chainChanged',
		listener: Web3Eip1193ProviderEventCallback<string>,
	): void {
		this._eventEmitter.on(type, listener);
	}
	public abstract on(
		type: 'accountsChanged',
		listener: Web3Eip1193ProviderEventCallback<string[]>,
	): void {
		this._eventEmitter.on(type, listener);
	}
	public abstract removeListener(
		type: string,
		listener: Web3Eip1193ProviderEventCallback<any> | Web3ProviderEventCallback,
	): void {
		this._eventEmitter.on(type, listener);
	}
```

And also doing the same for `once` and `removeListener`?

Reply:
It has been applied on `once` but not on `removeListener`. Is there a specific reason? Or was it just missed by a chance?